tests/integration/test_issue_2001.py

```diff
@@ -0,0 +1,37 @@
+import os.path
+import subprocess
+import sys
+
+from pex.testing import PY38, ensure_python_interpreter, make_env, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_compile_error_as_warning(
+    tmpdir,  # type: Any
+):
+    # type: (...) -> None
+    pex = os.path.join(str(tmpdir), "pex.pex")
+    run_pex_command(
+        args=[
+            "--include-tools",
+            "aenum",
+            "-o",
+            pex,
+        ],
+        # The package script requires Python 3.
+        python=sys.executable if sys.version_info[0] >= 3 else ensure_python_interpreter(PY38),
+    ).assert_success()
+    # The packaged Pex PEX should work with all Pythons we support, including the current test
+    # interpreter.
+    compiled_deps_dir = str(tmpdir)
+    output = subprocess.run(
```

Comment:
CI caught a real issue here actually: https://github.com/pantsbuild/pex/actions/runs/3688289216/jobs/6242918161#step:4:149

Pex supports back to Python 2.7 so the code must be 2.7 compatible. Just use:
```python
process = subprocess.check_call(args=..., env=..., stderr=...)
_, stderr_bytes = process.communicate()
...
```

Reply:
So it looks like this error isn't generated for python2.7
I even installed python2.7 locally to check and I don't see any warning being thrown

Reply:
Well, the issue was compiling Python 2.7 syntax under Python 3; not the other way around:
```
***   File "./pex-deps-compiled/lib/python3.9/site-packages/aenum/_py2.py", line 5
    raise exc, None, tb
             ^
SyntaxError: invalid syntax
```

So the test needs to check for no warning under Python 2.7 and warning otherwise.